### PR TITLE
Add new API "eigvals" in linalg

### PR DESCRIPTION
**Pull request by From00**
### PR types
New features

### PR changes
APIs

### Describe
Add a new API: linalg.eigvals

The corresponding PR for CN docs：https://github.com/PaddlePaddle/docs/pull/3875

### NOTE 
1. The gradient kernel of this operator does not yet develop.
2. This PR does not implement the CUDA kernel. If the API is called by a GPU tensor, Paddle will report an error.
3. This PR implements the linalg.eigvals API with the Eigen library, which will be replaced by the Lapack library in the future.

image
image

### CODE REVIEW

File: python/paddle/tensor/linalg.py [hunk 1]
```diff
@@ -1171,3 +1171,60 @@ def matrix_power(x, n, name=None):
...
+
+    Returns:
+        Tensor: A tensor containing the unsorted eigenvalues which has the same batch dimensions with `x`. 
+            The eigenvalues are complex-valued even when `x` is real.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
```
Comment (zhiqiu) [inline python/paddle/tensor/linalg.py:1202]: Add paddle.seed(x) to fix output.

File: python/paddle/fluid/tests/unittests/test_eigvals_op.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    vh = v.conj().T
+    print("vh:")
+    print(vh)
+    print("out_grad:")
+    print(out_grad)
+    a_grad = np.linalg.solve(vh, np.diagflat(out_grad, 0) * vh)
+    print("a_grad")
+    print(a_grad)
+
+    return a_grad.astype(a.dtype)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/test_eigvals_op.py]: seems no usage.

File: paddle/fluid/operators/eigvals_op.h [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+using EigenMatrixPaddle = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>;
+template <typename T>
+using EigenVectorPaddle = Eigen::Matrix<PaddleCType<T>, Eigen::Dynamic, 1>;
+template <typename T>
+using EigenMatrixStd =
+    Eigen::Matrix<StdCType<T>, Eigen::Dynamic, Eigen::Dynamic>;
+template <typename T>
+using EigenVectorStd = Eigen::Matrix<StdCType<T>, Eigen::Dynamic, 1>;
+
+static void SpiltBatchSquareMatrix(const Tensor *input,
```
Comment (zhiqiu) [inline paddle/fluid/operators/eigvals_op.h]: Better use const& for `input`

File: python/paddle/tensor/linalg.py (hunk 1 above)
Comment (From00) [reply python/paddle/tensor/linalg.py:1202]: done

File: python/paddle/fluid/tests/unittests/test_eigvals_op.py (hunk 2 above)
Comment (From00) [reply python/paddle/fluid/tests/unittests/test_eigvals_op.py]: It has been commented out.

File: paddle/fluid/operators/eigvals_op.h (hunk 3 above)
Comment (From00) [reply paddle/fluid/operators/eigvals_op.h]: done

**Review by zhiqiu — DISMISSED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LG API

**Review by XieYunshen — APPROVED** [review]
LGTM for set_tests_properties(test_eigvals_op PROPERTIES TIMEOUT 400)

**Review by TCChenlong — APPROVED** [review]
LGTM
TODO：
1、补充中文文档&中英文API文档预览图
2、示例代码中不需要set_device这一行

**Pull request merged**